app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt

```diff
@@ -69,45 +72,35 @@ class SettingPresenter(
 
     override fun onViewReady() {
         settingStore.onEnablingFingerprint
-            .subscribe {
-                if (it is FingerprintAuthAction.OnAuthentication) {
-                    when (it.authCallback) {
-                        is FingerprintAuthDialogFragment.AuthCallback.OnAuth -> dispatcher.dispatch(
-                            SettingAction.UnlockWithFingerprint(true)
-                        )
-                        is FingerprintAuthDialogFragment.AuthCallback.OnError -> {
-                            dispatcher.dispatch(SettingAction.UnlockWithFingerprint(false))
-                        }
-                    }
+            .map {
+                if (it is FingerprintAuthAction.OnAuthentication &&
+                    it.authCallback == FingerprintAuthDialogFragment.AuthCallback.OnAuth) {
```

Comment:
What happens in the case of `FingerprintAuthAction.OnCancel`?

Reply:
fixed this.